cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -253,29 +267,60 @@ where
 	async fn produce_bundle(self, slot_info: ExecutorSlotInfo) -> Option<BundleResult> {
 		println!("TODO: solve some puzzle based on `slot_info` to be allowed to produce a bundle");
 
-		let transactions = {
-			// selection policy: minimize the transaction equivocation.
-			println!("TODO: once elected, select unseen transactions from the transaction pool");
-			b"some transactions".to_vec()
+		let now = time::Instant::now();
+		// TODO: proper deadline
+		let deadline = now + time::Duration::from_micros(500);
+
+		// TODO: ready at the best number of primary block?
+		let parent_number = self.client.info().best_number;
+		let mut t1 = self.transaction_pool.ready_at(parent_number).fuse();
+		// TODO: proper timeout
+		let mut t2 = futures_timer::Delay::new(time::Duration::from_micros(100)).fuse();
+
+		let mut pending_iterator = select! {
+			res = t1 => res,
+			_ = t2 => {
+				tracing::warn!(
+					block_number,
+					"Timeout fired waiting for transaction pool, proceeding with production.",
+				);
+				self.transaction_pool.ready()
+			}
 		};
 
-		let _transactions_root = b"merkle root of transactions".to_vec();
+		// TODO: Select transactions properly from the transaction pool
+		//
+		// Selection policy:
+		// - minimize the transaction equivocation.
+		// - maximize the executor computation power.
+		let mut extrinsics = Vec::new();
+		while let Some(pending_tx) = pending_iterator.next() {
+			let now = std::time::Instant::now();
+			if now > deadline {
+				break
+			}
```

Comment:
This would be better:
```suggestion
		let start = std::time::Instant::now();
		// TODO: proper deadline
		let deadline = time::Duration::from_micros(500);
		while let Some(pending_tx) = pending_iterator.next() {
			if start.elapsed() >= deadline {
				break
			}
```